src/overlays/actors/ovl_En_M_Thunder/z_en_m_thunder.c

```diff
@@ -29,39 +39,533 @@ const ActorInit En_M_Thunder_InitVars = {
+    }
+}
+
+void EnMThunder_Spin_Attack(EnMThunder* this, PlayState* play) {
+    Player* player = GET_PLAYER(play);
+
+    if (Math_StepToF(&this->lightColorFrac, 0.0f, 0.0625f)) {
+        Actor_MarkForDeath(&this->actor);
+    } else {
+        Math_SmoothStepToF(&this->actor.scale.x, (s32)this->scaleTarget, 0.6f, 0.8f, 0.0f);
+        Actor_SetScale(&this->actor, this->actor.scale.x);
+        this->collider.dim.radius = this->actor.scale.x * 30.0f;
+        Collider_UpdateCylinder(&this->actor, &this->collider);
+        CollisionCheck_SetAT(play, &play->colChkCtx, &this->collider.base);
+    }
+
+    if (this->timer > 0) {
+        this->actor.world.pos.x = player->bodyPartsPos[0].x;
+        this->actor.world.pos.z = player->bodyPartsPos[0].z;
+        this->timer--;
+    }
+
+    if (this->lightColorFrac > (6.0f / 10.0f)) {
+        this->alphaFrac = 1.0f;
+    } else {
+        this->alphaFrac = this->lightColorFrac * (10.0f / 6.0f);
+    }
+
+    func_808B5EEC(this, play);
+
+    if (Play_InCsMode(play)) {
+        Actor_MarkForDeath(&this->actor);
+    }
+}
+
+void EnMThunder_SwordBeam_Attack(EnMThunder* this, PlayState* play) {
+    s32 pad[2];
+    f32 sp2C;
+
+    if (this->lightColorFrac > (9.0f / 10.0f)) {
+        this->alphaFrac = 1.0f;
+    } else {
+        this->alphaFrac = this->lightColorFrac * (10.0f / 9.0f);
+    }
+
+    if (Math_StepToF(&this->lightColorFrac, 0.0f, 0.05f)) {
+        Actor_MarkForDeath(&this->actor);
+    } else {
+        sp2C = -80.0f * Math_CosS(this->actor.world.rot.x);
+
+        this->actor.world.pos.x += sp2C * Math_SinS(this->actor.shape.rot.y);
+        this->actor.world.pos.z += sp2C * Math_CosS(this->actor.shape.rot.y);
+        this->actor.world.pos.y += -80.0f * Math_SinS(this->actor.world.rot.x);
+
+        Math_SmoothStepToF(&this->actor.scale.x, this->scaleTarget, 0.6f, 2.0f, 0.0f);
+        Actor_SetScale(&this->actor, this->actor.scale.x);
+
+        this->collider.dim.radius = this->actor.scale.x * 5.0f;
+
+        this->collider.dim.pos.x = this->actor.world.pos.x;
+        this->collider.dim.pos.y = this->actor.world.pos.y;
+        this->collider.dim.pos.z = this->actor.world.pos.z;
+
+        this->collider.dim.pos.x =
+            (Math_SinS(this->actor.shape.rot.y) * -5.0f * this->actor.scale.x) + this->actor.world.pos.x;
+        this->collider.dim.pos.y = this->actor.world.pos.y;
+        this->collider.dim.pos.z =
+            (Math_CosS(this->actor.shape.rot.y) * -5.0f * this->actor.scale.z) + this->actor.world.pos.z;
+
+        CollisionCheck_SetAT(play, &play->colChkCtx, &this->collider.base);
+    }
+
+    if (this->timer > 0) {
+        this->timer--;
+    }
+
+    func_808B5EEC(this, play);
+}
+
+void func_808B6310(EnMThunder* this, PlayState* play) {
+    if (Math_StepToF(&this->lightColorFrac, 0.0f, 0.0625f)) {
+        Actor_MarkForDeath(&this->actor);
+    } else {
+        Math_SmoothStepToF(&this->actor.scale.x, (s32)this->scaleTarget, 0.6f, 0.8f, 0.0f);
```

Comment:
```suggestion
        Math_SmoothStepToF(&this->actor.scale.x, this->scaleTarget, 0.6f, 0.8f, 0.0f);
```
It's not needed in other places, needed here?

Reply:
Unfortunately yes.